### ISSUE TITLE
[keymgr] SW can request EDN or cancel EDN req

### ISSUE THREAD

**Issue by weicaiyang** [lowRISC/opentitan#8360]
@tjaychen 
This is due to the update at #8322
Since EDN request relies on `reseed_interval_i`, SW can set it to 0 to send EDN request before the counter starts, or even cancel an unfinished EDN req as shown in the waveform, which violates the handshaking protocol.
`  assign edn_req = (reseed_cnt == reseed_interval_i);
`
![Screen Shot 2021-09-24 at 1 54 55 PM](https://user-images.githubusercontent.com/49293026/134738675-7a63aa44-d821-4c88-8844-9fa16229588f.png)

**Comment by tjaychen** [comment]
regarding the cancelling, shouldn't that be covered by the `regwen`? Ie, if we set the count and lock it, there should be no changing it?

the other 0 case doesn't look right, i'll have a look.

**Comment by weicaiyang** [comment]
> regarding the cancelling, shouldn't that be covered by the `regwen`? Ie, if we set the count and lock it, there should be no changing it?
> 
> the other 0 case doesn't look right, i'll have a look.

If it's locked, then it's fine. But lock is done by SW. If SW doesn't lock it, SW can request/cancel EDN. I feel SW should only set the interval, and should not have capability to directly control EDN in any way.

**Comment by tjaychen** [comment]
yeah but this basically controls the request interval though.. i'm not sure
it's worth putting in more logic to guard it.
Because I think the way it'll be used is simply..program/lock and go.

Otherwise we'll have to go down a road of..
when edn makes the request, latch the current reseed value so that it
cannot change, and only allow a change later.
That'll require extra storage that I'm really not sure is worth it.

Although having said that, i've been meaning to switch this to the
prim_edn_req_ack module, maybe that will handle this natively.
Let me take a look (i did this part before that prim module existed).
